### ISSUE TITLE
Transformers scheduler doesn't alter LR of added param group after model unfreeze

### ISSUE THREAD

**Issue by rbracco** [huggingface/transformers#20029]
### System Info

python 3.8.10 on ubuntu 20.04
pytorch 1.12.1
transformers 4.20.1

### Who can help?

@sgugger 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

1. Add a transformers scheduler, such as `transformers.get_linear_schedule_with_warmup` to any training run that begins with a frozen model and unfreezes. 
2. Monitor the learning rate, LR for param groups that are unfrozen after the 0th epoch are constant and unaffected by the scheduler

Sorry I don't have a better reproduction section. I mainly use transformers as a dependency of another library, and I tried making a reproducible script [here on Colab](https://colab.research.google.com/drive/1LXkIUP_vcVV3Vmc0sLInpIUEWU2H8K6P?usp=sharing), but Colab is crashing due to one of the imports. 

### Expected behavior

I believe this is a bug but it could be expected behavior. I would expect the unfrozen param groups (param-group 2) should also be controlled by the scheduler, as they are when using pytorch schedulers such as `torch.optim.lr_scheduler.LinearLR`. 

### LR of param groups after unfreeze when using `torch.optim.lr_scheduler.LinearLR`.
![image](https://user-images.githubusercontent.com/47190785/199602432-e47433b3-9a09-4d89-b379-dc6f8a0e9877.png)

### LR of param groups after unfreeze when using `transformers.get_linear_LR_with_warmup`.
![image](https://user-images.githubusercontent.com/47190785/199602498-0bde2ea8-ca12-439c-8add-35d92d35e7e5.png)

### LR of param groups after unfreeze when using `transformers.get_cosine_schedule_with_warmup`
![image](https://user-images.githubusercontent.com/47190785/199602591-1ca3e075-1600-46df-9bf1-ae3b99daca4b.png)

**Comment by sgugger** [comment]
The Trainer by itself does not support several parameter groups, you will need to subclass and overwrite the methods that create schedulers/optimizers.

**Comment by rbracco** [comment]
Thank you!
